### ISSUE TITLE
Go testcases run failed for ziplist in CI

### ISSUE THREAD

**Issue by PragmaTwice** [apache/kvrocks#868]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Version

unstable e8efdf28260d31f181fc200ec9082f36d9294c47

### Minimal reproduce step

refer to https://github.com/apache/incubator-kvrocks/runs/8303808021

```
    list_test.go:202: 
        	Error Trace:	/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/type/list/list_test.go:202
        	Error:      	Not equal: 
        	            	expected: 200
        	            	actual  : 201
        	Test:       	TestZipList/ziplist_implementation:_value_encoding_and_backlink
=== RUN   TestZipList/ziplist_implementation:_encoding_stress_testing
    list_test.go:229: 
        	Error Trace:	/Users/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/type/list/list_test.go:229
        	Error:      	Not equal: 
        	            	expected: 282
        	            	actual  : 283
        	Test:       	TestZipList/ziplist_implementation:_encoding_stress_testing
--- FAIL: TestZipList (18.74s)
```

### What did you expect to see?

success

### What did you see instead?

fail

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by tisonkun** [comment]
cc @torwig It seems this is an unstable test. Could you try to dig it out?

**Comment by git-hulk** [comment]
Not sure whether is caused by Go Redis client retry (default is 3) or not, see: [options.go#L63](https://github.com/go-redis/redis/blob/3d1e2e5bc8af1bfec675334a9a3f4cd9919dfda3/options.go#L63). Can disable the retry first if we don't know how to reproduce.

**Comment by tisonkun** [comment]
@git-hulk OK. I made a patch #869 and let's see whether it works.

**Comment by tisonkun** [comment]
Closed as #869 merged. Feel free to reopen if it occurs again.
